src/web4/handler.rs

```diff
@@ -14,16 +14,59 @@ use crate::{
 };
 
 pub const BASE64_ENGINE: engine::GeneralPurpose =
-    engine::GeneralPurpose::new(&alphabet::URL_SAFE, general_purpose::NO_PAD);
+    engine::GeneralPurpose::new(&alphabet::URL_SAFE, general_purpose::PAD);
 
 pub fn web4_get(contract: &Contract, request: Web4Request) -> Web4Response {
     let path_parts: Vec<&str> = request.path.split('/').collect();
 
     // A valid path provided by a legit web4 gateway always has '/', so there
     // are always [0] and [1] elements, and [0] is always empty.
     let page = path_parts[1];
+
+    let metadata_preload_url = format!("https://rpc.web4.near.page/account/social.near/view/get?keys.json=[%22{}/widget/app/metadata/**%22]", env::current_account_id());
```

Comment:
You don't need to query specific RPC. Use `/web4/contract/...` API provided by web4 gateway:

<img width="902" alt="image" src="https://github.com/NEAR-DevHub/neardevhub-contract/assets/304265/53506d23-2c48-4073-94ad-8db83fbd55b3">

If you have any further questions, ask them in https://t.me/web4near

Reply:
thanks! I should have seen this from the docs.